### PR TITLE
Fix relation fields in indexes in datamodel converter

### PR DESCRIPTION
**Pull request by tomhoule**
The crash was happening when building the Index from the IndexTemplate.

### CODE REVIEW

**Comment by tomhoule** [comment]
@pimeys  do you want to review this? I tried to avoid touching the query engine as much as possible and just fix the datamodel converter, since that code is probably going to change very soon.

**Comment by tomhoule** [comment]
This should fix https://github.com/prisma/prisma2/issues/1377

File: libs/prisma-models/src/field.rs [hunk 1]
```diff
@@ -20,6 +20,30 @@ pub enum Field {
     Scalar(ScalarFieldRef),
 }
 
+impl Field {
```
Comment (pimeys) [inline libs/prisma-models/src/field.rs]: These will be completely changed in Dom's branch...

File: libs/prisma-models/src/index.rs [hunk 2]
```diff
@@ -9,18 +10,18 @@ pub struct IndexTemplate {
...
             IndexType::Unique => self
                 .fields
                 .into_iter()
                 .map(|name| {
                     let field = fields
-                        .iter()
-                        .find(|sf| sf.name == name)
-                        .expect(&format!("Unable to resolve scalar field '{}'", name));
+                        .find_from_all(&name)
+                        .with_context(|err| format!("Unable to resolve scalar field '{}'. {}", name, err))
```
Comment (pimeys) [inline libs/prisma-models/src/index.rs]: :+1:

File: query-engine/core/src/query_graph_builder/extractors/filters.rs [hunk 3]
```diff
@@ -226,7 +226,7 @@ fn find_index_fields(name: &str, model: &ModelRef) -> QueryGraphBuilderResult<Ve
         .unique_indexes()
         .into_iter()
         .find(|index| &compound_field_name(index) == name)
-        .map(|index| index.fields())
+        .map(|index| index.scalar_fields())
```
Comment (pimeys) [inline query-engine/core/src/query_graph_builder/extractors/filters.rs]: I don't understand this... So what on earth we are doing here? Maybe I don't understand the function, but shouldn't we just filter all fields now for the index?

File: query-engine/core/src/schema_builder/input_type_builder/mod.rs [hunk 4]
```diff
@@ -126,7 +126,7 @@ pub trait InputTypeBuilderBase<'a>: CachedBuilder<InputObjectType> + InputBuilde
     /// Generates and caches an object type for a unique index.
     fn compound_field_unique_object_type(&self, index: &Index) -> InputObjectTypeRef {
         let name = index.name.as_ref().map(|n| capitalize(n)).unwrap_or_else(|| {
-            let index_fields = index.fields();
+            let index_fields = index.scalar_fields();
```
Comment (pimeys) [inline query-engine/core/src/schema_builder/input_type_builder/mod.rs]: Same as above.

File: query-engine/core/src/schema_builder/input_type_builder/mod.rs [hunk 5]
```diff
@@ -138,7 +138,7 @@ pub trait InputTypeBuilderBase<'a>: CachedBuilder<InputObjectType> + InputBuilde
         let input_object = Arc::new(init_input_object_type(name.clone()));
         self.cache(name, Arc::clone(&input_object));
 
-        let index_fields = index.fields();
+        let index_fields = index.scalar_fields();
```
Comment (pimeys) [inline query-engine/core/src/schema_builder/input_type_builder/mod.rs]: Same as above.

File: query-engine/core/src/schema_builder/utils.rs [hunk 6]
```diff
@@ -136,7 +136,7 @@ pub fn append_opt<T>(vec: &mut Vec<T>, opt: Option<T>) {
 /// Computes a compound field name based on an index.
 pub fn compound_field_name(index: &Index) -> String {
     index.name.clone().unwrap_or_else(|| {
-        let index_fields = index.fields();
+        let index_fields = index.scalar_fields();
```
Comment (pimeys) [inline query-engine/core/src/schema_builder/utils.rs]: Same as above.

**Comment by tomhoule** [comment]
Talked with @mavilein about it - this will be completely overwritten by the AUMFIDARR work, so I'll close this PR.

**Pull request closed without merging**